### ISSUE TITLE
[no-cycle] Should report error even if no bindings are specified

### ISSUE THREAD

**Issue by latin-1** [import-js/eslint-plugin-import#2696]
We may need a stricter version of rule `no-cycle`.

### Case 1: Error

```js
// a.js
import "./b";
```

```js
// b.js
import "./a";
```

### Case 2: Error

```js
// a.js
import {} from "./b";
```

```js
// b.js
import {} from "./a";
```

### Case 3: Error

```ts
// a.ts
import {} from "./b";
```

```ts
// b.ts
import { type A } from "./a";
```

### Case 4: OK

```ts
// a.ts
import {} from "./b";
```

```ts
// b.ts
import type { A } from "./a";
```

**Comment by ljharb** [comment]
If there's no bindings, why is the cycle a problem?

**Comment by latin-1** [comment]
After some thought, I realize you are correct. There are no particular use cases. Sorry for bothering you.
